### PR TITLE
[dashboard][api] Fix, PUT publish/draft to not clean slug and owners

### PR DESCRIPTION
**Pull request by dpgaspar**
### CATEGORY

- [X] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Fix bug on the dashboard API, "Draft" Button Clears Slug and Owners from Dashboard.

Note: We are planning to refactor this soon to comply to SIP-35

### TEST PLAN
Create a new dashboard with a slug and several owners, then publish the dashboard using the dashboard view itself. Verify that slug and owner do not get cleaned  

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/dashboard/api.py [hunk 1]
```diff
@@ -88,12 +88,14 @@ class BaseDashboardSchema(BaseOwnedSchema):
...
         super().pre_load(data)
-        data["slug"] = data.get("slug")
-        data["owners"] = data.get("owners", [])
-        if data["slug"]:
-            data["slug"] = data["slug"].strip()
-            data["slug"] = data["slug"].replace(" ", "-")
-            data["slug"] = re.sub(r"[^\w\-]+", "", data["slug"])
+        if "slug" in data:
+            data["slug"] = data.get("slug")
+            if data["slug"]:
```
Comment (villebro) [inline superset/views/dashboard/api.py]: If `"slug" in data`, aren't the next rows unnecessary? Seems like we could just replace these three rows with `if get("slug"):`

**Review by villebro — CHANGES_REQUESTED** [review]
I think this can be simplified slightly?

File: superset/views/dashboard/api.py (hunk 1 above)
Comment (dpgaspar) [reply superset/views/dashboard/api.py]: not exactly, I'm need to make sure that:
- Exists a slug key?
- If the slug key exists does it have data?
- If the slug key exists and it has data, apply string tranformations

Comment (dpgaspar) [reply superset/views/dashboard/api.py]: Forget it! your right

**Review by villebro — APPROVED** [review]

File: superset/views/dashboard/api.py [hunk 2]
```diff
@@ -88,12 +88,12 @@ class BaseDashboardSchema(BaseOwnedSchema):
...
         super().pre_load(data)
-        data["slug"] = data.get("slug")
-        data["owners"] = data.get("owners", [])
-        if data["slug"]:
+        if data.get("slug"):
             data["slug"] = data["slug"].strip()
             data["slug"] = data["slug"].replace(" ", "-")
             data["slug"] = re.sub(r"[^\w\-]+", "", data["slug"])
+        if "owners" in data:
+            data["owners"] = data.get("owners", [])
```
Comment (john-bodley) [inline superset/views/dashboard/api.py]: Why are lines 95 - 96 required? If the `owners` key is in `data` then `data.get("owners", [])` is equivalent to `data["owners"]` and thus line 96 becomes `data["owners"] = data["owners"]` which is unnecessary.

Comment (villebro) [reply superset/views/dashboard/api.py]: I think this is to make sure that `owners` is always a list if the key is present. With the help of these two rows, if `data == {"owners": None}`, `data` becomes `{"owners": []}`, but doesn't get set if `data` doesn't contain the key `owners`.

Comment (john-bodley) [reply superset/views/dashboard/api.py]: @villebro per the [`dict.get`](https://docs.python.org/3/library/stdtypes.html#dict.get) docs, the default is only returned if he key does not exist. I think the desired logic should be, 

```python
if "owners" in data and data["owners"] is None:
  data["owners"] = []
```

Comment (villebro) [reply superset/views/dashboard/api.py]: I agree @john-bodley , this is much more intuitive 👍

Comment (dpgaspar) [reply superset/views/dashboard/api.py]: yes, it's better, changed it

**Comment by milimetric** [comment]
We're seeing an issue and it seems related to this change, but I'd like to verify.  So the behavior we're seeing is:

* User makes a new dashboard
* Clicks Draft button
* (sometimes) is prompted to log in
* (sometimes) is not prompted to log in
* (always) gets a 401 error with the message {"message":"Access is Denied","severity":"danger"}

Would this be fixed after we get this change?

**Comment by milimetric** [comment]
Gengle ping, can @dpgaspar or anyone else chime in?

**Comment by milimetric** [comment]
UPDATE: this issue went away when we upgraded to version 0.36

**Pull request merged**